test/components/form/inputs_test.exs

```diff
@@ -0,0 +1,69 @@
+defmodule Surface.Components.Form.InputsTest do
+  use ExUnit.Case, async: true
+
+  alias Surface.Components.Form, warn: false
+  alias Surface.Components.Form.Inputs, warn: false
+  alias Surface.Components.Form.TextInput, warn: false
+
+  import ComponentTestHelper
+
+  test "using generated form received as slot props" do
+    code = """
+    <Form for={{ :parent }} opts={{ csrf_token: "test" }}>
+      <Inputs for={{ :children }} :let={{ form: f }}>
```

Comment:
I wonder if users might be confused between `for` and `:for`.

Reply:
For the `<Inputs>` component we could just call it `field` but for the `<Form>` component I'm not sure. Should we maybe use separate properties? Like: 

```jsx
<Form changeset={{ @changeset }}>
and
<Form name={{ :my_form }}>
```

Reply:
A separate discussion is whether we should use `<FormFor>` and `<InputsFor>` instead. 

Although using `<FormFor>` and `<InputsFor>` would be more intuitive for people already using Phoenix. I find them much less intuitive for other users. Besides, since we need to have a property like `for` to pass the value (as opposed to just pass it as an unnamed argument like in `form_for`), it seems to make sense to use just `<Form>`.

Reply:
> I wonder if users might be confused between `for` and `:for`.

I could see that, though I think the context for using and describing the two is different enough that it's reasonable - i.e. `for` makes sense as a property name here, and that documentation is associated with the `Inputs` and `Form` components

> Should we maybe use separate properties?

🤔 if we were using separate properties, that would open users up to potentially specifying two different values for the same input. (i.e. `<Form name={{ :my_form }} changeset={{ @changeset }}>`). I prefer a single property so we don't even have to consider how to validate that.

> using `<FormFor>` and `<InputsFor>` would be more intuitive for people already using Phoenix

maybe more familiar or a better `1:1` mapping, but i would say that `<Form for={{ @changeset }}>` reads a bit better than `<FormFor for={{ @changeset }}>`; and if we stick with the property named `for`, keeps that fairly close mapping. imo the reason `form_for` works is because the first argument is the thing the form is for, so both read the same (at least to me)